user_interface.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <string>
+#include "user_interface.h"
+#include <cmath>
+#include <iomanip> //std::setprecision
+
+using namespace std;
+//const ostream &standard = std::cout;
+const string white_line = "\011\012\013\014\015\040";
+
+void print_greeting(ostream &out)
+{
+    out << "Welcome to the random number generator!\n"
+        << "It was done as a laboratory programming job.\n";
+}
+
+string clear_whites(const string &s){
+    size_t first = s.find_first_not_of(white_line);
+
+    if(first == string::npos) return "";
+    size_t last = s.find_last_not_of(white_line, s.size() );
+    return s.substr(first, last - first + 1);
+}
+
+int choose_gen(ostream &out)
+{
+    bool correct_ans = false;
+    string ans = "";
+    int res = 11;
+    size_t ans_len = 0;
+    do{
+        print_help();
+        out << "Select one of the generators by entering its serial number.\n"
+            << "Enter H to see help again.\n"
+            << "Enter Q to exit the program.\n"
+            << "Hint: All other symbols will call help info and ask you to enter the data again.\n" ;
+        cin >> ans;
+        if(cin.fail()){
+            cin.clear();
+            cin.ignore();
+            throw out_of_range("***** FAILED TO GET STRING FROM STD::CIN *****\nWatch size_t choose_gen(ostream &out) to solve the problem\n");
+        }
+        ans = clear_whites(ans);
+        ans_len = ans.length();
+        if(ans_len > 0 && ans_len < 3){
+            if(ans_len == 2 && ans == "10"){
```

Comment:
It's redundant to check length if you check equality. Also, you can return right here because you know you've found a valid input. 
```suggestion
            if (ans == "10") {
                return 10;
            }
```